hazelcast/src/test/java/com/hazelcast/cache/CacheCreationTest.java

```diff
@@ -80,6 +82,45 @@ public void createSingleCache() {
         cache.get(1);
     }
 
+    @Test
+    public void concurrentCacheCreation() throws InterruptedException {
+        // see https://github.com/hazelcast/hazelcast/issues/17284
+        String cacheName = "myCache";
+        int threadCount = Runtime.getRuntime().availableProcessors() * 20;
```

Comment:
20 looks like a good number:)

this is a slow serial test, having more threads than cores should cause enough context switching to trigger the race condition. On my laptop the test fails in +- 50% of runs.